File: app/src/main/java/com/gateway/android/GatewayApplication.java
package com.gateway.android;

import android.app.Application;
import com.google.firebase.messaging.FirebaseMessaging;

public class GatewayApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();

        if (BuildConfig.DEBUG) {
            FirebaseMessaging.getInstance().subscribeToTopic("gateway-debug");
        } else {
            FirebaseMessaging.getInstance().subscribeToTopic("gateway");
        }

        //com.coingapp.android.util.SharedPreferenceWrapper.getInstance().initialize(this);
    }
}

File: app/src/main/java/com/gateway/android/GatewayApplication.kt
package com.gateway.android

import android.app.Application
import android.telephony.TelephonyManager
import com.gateway.android.utils.SharedPreferencesWrapper
import com.gateway.android.utils.Util

class GatewayApplication : Application() {

    override fun onCreate() {
        super.onCreate()

        val telephonyManager = getSystemService(TELEPHONY_SERVICE) as TelephonyManager

        SharedPreferencesWrapper.getInstance(this).simState = telephonyManager.simState
        Util.checkConnectivity(this)
    }
}
